### PR TITLE
Added buy and sell for AMM. Deployer is now a task. README updated

### LINKED ISSUE #10: Move deployer to tasks
Hardhat [tasks](https://hardhat.org/guides/create-task.html) are probably what we want instead of scripts. We should write tasks in their own directory and import them into hardhat.config.ts. The only script we have so far is the deployer so start there.

The deployer should lose the "commander" code and use hardhat's task integration.

There may be some difficulty with hardhat's configuration vs our own monolithic configuration. Verify that the task works correctly by deploying to kovan.

### LINKED ISSUE #11: Balancer pool
- [x] create balancer pool for market share tokens - #12 
- [x] wrapper for trading collateral to/from shares
- [ ] wrapper for liquidity, collateral to/from shares

### LINKED ISSUE #13: Deployment script doesn't recognize `--network` flag; Kovan deployment doesn't work
Short version: the custom deployment scripts use an `--environment` instead of `--network` parameter, which puts them in disharmony with the readme as well as hardhat.  Kovan deployment also isn't working.

Long version:

The readme says to run this command to deploy to the local hardhat chain:
```
yarn smart contracts:deploy --network localhost
```
However, the `--network` flag isn't recognized.  Instead, it is expecting an [`--environment` flag](https://github.com/AugurProject/turbo/blob/master/packages/smart/scripts/deploy-turbo.ts#L20), with argument [`local`](https://github.com/AugurProject/turbo/blob/master/packages/smart/src/environments/index.ts#L6-L9) instead of `localhost`.  When I run `yarn smart contracts:deploy --environment local`, it deploys successfully.

However, I can't get the contracts deployed to Kovan at all.  I tried the following (again, it expects `environment` instead of `network`):
```bash
jack@substrate:~/src/turbo$ PRIVATE_KEY=<0xwhatever> yarn smart contracts:deploy --environment kovan
yarn run v1.22.5
$ yarn workspace @turbo/smart contracts:deploy --environment kovan
$ ts-node --files ./scripts/deploy-turbo.ts --environment kovan
Error: unknown account #0 (operation="getAddress", code=UNSUPPORTED_OPERATION, version=providers/5.0.24)
    at Logger.makeError (/home/jack/src/turbo/node_modules/@ethersproject/logger/src.ts/index.ts:205:28)
    at Logger.throwError (/home/jack/src/turbo/node_modules/@ethersproject/logger/src.ts/index.ts:217:20)
    at /home/jack/src/turbo/node_modules/@ethersproject/providers/src.ts/json-rpc-provider.ts:149:24
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at Function.create (/home/jack/src/turbo/node_modules/@nomiclabs/hardhat-ethers/src/signers.ts:5:34)
    at main (/home/jack/src/turbo/packages/smart/scripts/deploy-turbo.ts:36:18) {
  reason: 'unknown account #0',
  code: 'UNSUPPORTED_OPERATION',
  operation: 'getAddress'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 1
Command: /home/jack/.nvm/versions/node/v14.16.0/bin/node
Arguments: /usr/share/yarn/lib/cli.js contracts:deploy --environment kovan
Directory: /home/jack/src/turbo/packages/smart
Output:
```
It's not clear to me if the Kovan deployment issue is caused by the flag confusion, or is something different.  (Or, maybe I'm just messing something up...?)

I have a small suggestion and a larger one.  The small suggestion is just to make sure that your readme is harmonized with your code.  The larger suggestion (from the perspective of someone who knows the pain of maintaining custom deployment machinery), is to avoid maintaining custom deployment machinery, if it's at all possible.  For example, if there's not a compelling reason to use `--environment` over `--network` (and `local` over `localhost`), use `--network` and `localhost` so that the flags match the underlying hardhat flags.  Similarly, is there anything in `src/environments` that couldn't just be specified in the hardhat config file?  (I see the kovan endpoint hardcoded both in the config file and in `src/environments/kovan.ts`...)

### PR DESCRIPTION
**Pull request by robert-e-davidson3**
This does NOT test selling shares to the AMM for collateral. That depends on #18

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,12 +25,16 @@
...
     "chai": "^4.3.4",
     "eslint": "^7.22.0",
     "eslint-config-prettier": "^8.1.0",
     "ts-generator": "^0.1.1",
     "ts-node": "^9.1.1",
     "typescript": "^4.2.3"
+  },
+  "dependencies": {
+    "eslint-plugin-promise": "^4.3.1",
+    "remove": "^0.1.5"
```
Comment (justinbarry) [inline package.json:38]: Where is this used?

Comment (robert-e-davidson3) [reply package.json:38]: balls. I tried to remove a package but added the remove package...

Comment (robert-e-davidson3) [reply package.json:38]: also those should be in dev

Comment (robert-e-davidson3) [reply package.json]: fixed. good catch!

File: packages/smart/src/deployer.ts [hunk 2]
```diff
@@ -200,9 +181,9 @@ export class TurboCreator {
     specified: Partial<TrustedArbiterConfiguration> & { arbiter: string }
   ): TrustedArbiterConfiguration {
     return {
-      creatorFee: ethers.BigNumber.from(10).pow(16),
+      creatorFee: BigNumber.from(10).pow(16),
       outcomeSymbols: ["NO", "YES"],
-      outcomeNames: ["NO", "YES"].map(ethers.utils.formatBytes32String),
+      outcomeNames: ["NO", "YES"].map(utils.formatBytes32String),
```
Comment (bthaile) [inline packages/smart/src/deployer.ts:186]: Need to start using outcome name constants so we can be consistent across the repo

**Review by bthaile — APPROVED** [review]
lgtm

File: packages/smart/src/deployer.ts (hunk 2 above)
Comment (robert-e-davidson3) [reply packages/smart/src/deployer.ts:186]: good point

**Pull request merged**